testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/iap/IAPExecutor.kt

```diff
@@ -0,0 +1,101 @@
+package com.rakuten.tech.mobile.testapp.ui.iap
+
+import android.app.Activity
+import android.util.Log
+import com.android.billingclient.api.*
+import com.rakuten.tech.mobile.testapp.helper.showAlertDialog
+
+class IAPExecutor(val activity: Activity) {
```

Comment:
Do you think we could set this up similar to the way we setup Ad Mob? I'm worried that the current implementation leaves a lot of duplicate work which needs to be done by the Host App.

For example, we could create a separate module for `in-app-purchase` (similar to the `admob-latest` module) and then create a default implementation class of `InAppPurchaseBridgeDispatcher` which wraps some of the functionality of `IAPExecutor`. Then the Host App just needs to create an instance of the Default InAppPurchaseBridgeDispatcher.

Reply:
Yeah, this is a good idea with having a separate module with those default implementation. I'm looking forward to those changes.